packages/server/api/typing.ts

```diff
@@ -1,7 +1,18 @@
+import type {TForum, TTopic, TMessage, TMessageReaction} from 'server/api/models';
+
 type TForumAction = 'forum.list' | 'forum.create' | 'forum.delete';
 type TTopicAction = 'topic.list' | 'topic.create' | 'topic.delete';
 type TMessageAction = 'message.list' | 'message.create' | 'message.delete';
+
+export type TApiAction = TForumAction & TTopicAction & TMessageAction;
+export type TApiData = TForum & TTopic & TMessage & TMessageReaction;
+
 export type TPostData = {
-	action: TForumAction | TTopicAction | TMessageAction;
-	data: object;
+	action: TApiAction;
+	data: TApiData;
+};
+
+export type TApiResponseData = {
+	reason?: string;
+	data?: object;
```

Comment:
Сюда прямо проситься дженерик, чтобы удобно расставить возвращаемые типы:
```suggestion
export type TApiResponseData<T extends object> = {
	reason?: string;
	data?: T;
```

Это рекомендательный комментарий, если не считаешь необходимым можно не реализовывать